php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1479 @@
+      f.compilationUnit(b.optional(SCRIPT()), b.optional(b.token(PHPLexicalGrammar.SPACING)), b.token(PHPLexicalGrammar.EOF)));
+  }
+
+  public ScriptTree SCRIPT() {
+    return b.<ScriptTree>nonterminal(PHPLexicalGrammar.SCRIPT).is(
+      f.script(b.token(PHPLexicalGrammar.FILE_OPENING_TAG), b.zeroOrMore(TOP_STATEMENT())));
+  }
+
+  /**
+   * [ START ] Declaration
+   */
+
+  public VariableDeclarationTree MEMBER_CONST_DECLARATION() {
+    return b.<VariableDeclarationTree>nonterminal(PHPLexicalGrammar.MEMBER_CONST_DECLARATION).is(
+      f.memberConstDeclaration(
+        b.token(PHPLexicalGrammar.IDENTIFIER),
+        b.optional(f.newTuple18(b.token(EQU), STATIC_SCALAR()))));
+  }
+
+  public VariableDeclarationTree CONST_VAR() {
+    return b.<VariableDeclarationTree>nonterminal(PHPLexicalGrammar.CONSTANT_VAR).is(
+      f.constDeclaration(
+        b.token(PHPLexicalGrammar.IDENTIFIER),
+        b.token(EQU),
+        STATIC_SCALAR()));
+  }
+
+  public VariableDeclarationTree VARIABLE_DECLARATION() {
+    return b.<VariableDeclarationTree>nonterminal(PHPLexicalGrammar.VARIABLE_DECLARATION).is(
+      f.variableDeclaration(
+        b.token(PHPLexicalGrammar.REGULAR_VAR_IDENTIFIER),
+        b.optional(f.newTuple96(b.token(EQU), STATIC_SCALAR()))));
+  }
+
+  public NamespaceNameTree NAMESPACE_NAME() {
+    return b.<NamespaceNameTree>nonterminal(PHPLexicalGrammar.NAMESPACE_NAME).is(
+      b.firstOf(
+        f.namespaceName(
+          b.optional(b.token(PHPPunctuator.NS_SEPARATOR)),
+          b.zeroOrMore(f.newTuple4(
+            b.firstOf(b.token(PHPLexicalGrammar.IDENTIFIER)),
+            b.token(PHPPunctuator.NS_SEPARATOR)
+          )),
+          b.token(PHPLexicalGrammar.IDENTIFIER)),
+        f.namespaceName(
+          b.token(PHPKeyword.NAMESPACE),
+          b.token(PHPPunctuator.NS_SEPARATOR),
+          b.zeroOrMore(f.newTuple6(
+            b.firstOf(b.token(PHPLexicalGrammar.IDENTIFIER)),
+            b.token(PHPPunctuator.NS_SEPARATOR)
+          )),
+          b.token(PHPLexicalGrammar.IDENTIFIER))));
+  }
+
+  public UseClauseTree USE_CLAUSE() {
+    return b.<UseClauseTree>nonterminal(PHPLexicalGrammar.USE_CLAUSE).is(
+      f.useClause(
+        NAMESPACE_NAME(),
+        b.optional(
+          f.newTuple91(
+            b.token(PHPKeyword.AS),
+            b.token(PHPLexicalGrammar.IDENTIFIER)))));
+  }
+
+  public ClassDeclarationTree CLASS_DECLARATION() {
+    return b.<ClassDeclarationTree>nonterminal(PHPLexicalGrammar.CLASS_DECLARATION).is(
+      f.classDeclaration(
+        b.optional(b.firstOf(b.token(ABSTRACT), b.token(FINAL))),
+        b.token(CLASS),
+        IDENTIFIER(),
+        b.optional(f.newTuple50(b.token(EXTENDS), NAMESPACE_NAME())),
+        b.optional(f.newTuple30(b.token(IMPLEMENTS), INTERFACE_LIST())),
+        b.token(LCURLYBRACE),
+        b.zeroOrMore(CLASS_MEMBER()),
+        b.token(RCURLYBRACE)));
+  }
+
+  public ClassDeclarationTree TRAIT_DECLARATION() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)